### PR TITLE
Add TravisCI config file

### PR DESCRIPTION
**Pull request by jbarr21**
- Adding TravisCI config to address #4 
- Fix a bunch of lints in the sample app
- Bump support lib and leak canary in sample app
- Print lint reports to standard output on CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+android:
+  components:
+    - tools
+    - android-25
+    - build-tools-25.0.1
+    - platform-tools
+    - extra-android-m2repository
+
+after_failure:
+    - cat $TRAVIS_BUILD_DIR/viewpump/build/reports/lint-results.xml || true
```
Comment (hzsweers) [inline .travis.yml]: Nice. You can also selectively enable lintOptions to output to stdout like [this](https://github.com/hzsweers/empty-android-project/blob/master/app/build.gradle#L77-L87)

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+language: android
+
+jdk: oraclejdk8
+
+android:
+  components:
+    - tools
+    - android-25
+    - build-tools-25.0.1
```
Comment (hzsweers) [inline .travis.yml]: I think we're up to 25.0.3

File: ViewPumpSample/build.gradle [hunk 3]
```diff
@@ -23,7 +23,7 @@ android {
     }
 
     lintOptions {
-        disable 'AppCompatCustomView'
+        abortOnError false
```
Comment (hzsweers) [inline ViewPumpSample/build.gradle]: Can you add a comment explaining this?

File: viewpump/build.gradle [hunk 4]
```diff
@@ -18,6 +18,10 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'consumer-proguard-rules.pro'
         }
     }
+
+    lintOptions {
+        disable 'GradleDependency'
```
Comment (hzsweers) [inline viewpump/build.gradle]: Can you add a comment explaining this?

File: ViewPumpSample/build.gradle (hunk 3 above)
Comment (jbarr21) [inline ViewPumpSample/build.gradle]: This is just the sample app, we don't care about lint warnings here. Most of them were things like GradleDependency or CustomView

File: viewpump/build.gradle (hunk 4 above)
Comment (jbarr21) [inline viewpump/build.gradle]: this ignores the check since we are not using the latest support library

File: .travis.yml (hunk 2 above)
Comment (jbarr21) [inline .travis.yml]: thats for platform tools. 25.0.1 is the latest for build tools

File: .travis.yml (hunk 1 above)
Comment (jbarr21) [inline .travis.yml]: oh that's cool! guess i would've known that if i used gradle `3.4.1-all` :smiley:

File: ViewPumpSample/build.gradle [hunk 5]
```diff
@@ -23,18 +23,20 @@ android {
...
+        textOutput 'stdout'
+        textReport System.getenv('CI') == 'true'
     }
 }
 
 dependencies {
     compile project(':viewpump')
-    compile 'com.android.support:support-v4:25.0.1'
-    compile 'com.android.support:appcompat-v7:25.0.1'
+    compile 'com.android.support:support-v4:25.2.0'
```
Comment (hzsweers) [inline ViewPumpSample/build.gradle]: Can be 25.3.0 now

File: ViewPumpSample/build.gradle [hunk 6]
```diff
@@ -23,18 +23,20 @@ android {
     }
 
     lintOptions {
-        disable 'AppCompatCustomView'
+        // we don't need to be indexed by firebase and we don't have a custom icon for our sample app
+        disable 'GoogleAppIndexingWarning', 'MissingApplicationIcon'
+        textOutput 'stdout'
+        textReport System.getenv('CI') == 'true'
```
Comment (hzsweers) [inline ViewPumpSample/build.gradle]: I'm not sure if Travis sets this automatically, make sure to add it when we turn it on

Comment (jbarr21) [inline ViewPumpSample/build.gradle]: it does: [link](https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables)

Comment (hzsweers) [inline ViewPumpSample/build.gradle]: NICE

File: ViewPumpSample/build.gradle (hunk 5 above)
Comment (chrisjenx) [inline ViewPumpSample/build.gradle]: Bump

**Pull request merged**